### PR TITLE
Allow proxyListenPort to configure ip-table rules

### LINKED ISSUE #24169: IstioOperator MeshConfig.proxyListenPort doesn't configure the iptables redirection
**Bug description**
Configuring IstioOperator MeshConfig.proxyListenPort to a value other than 15001 does not take effect because the istio-iptables do not use this value. The injection-template has 15001 hardcoded instead of reading from MeshConfig.proxyListenPort (https://github.com/istio/istio/blob/85075d3e22422c48da103991945ae2f10c46c1e4/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L18)

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Expected MeshConfig.proxyListenPort to also configure the port in istio-iptables args.

**Steps to reproduce the bug**
1. Set proxyListenPort to something other than 15001 via IstioOperator spec.
2. Run istioctl command and check the injection-template.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
$ istioctl version --remote
no running Istio pods in "istio-system"
e12a938f76ea48aae85c63f17609b732295bcd24-dirty
```

**How was Istio installed?**
Via Operator

**Environment where bug was observed (cloud vendor, OS, etc)**
Any k8s cluster.

### PR DESCRIPTION
**Pull request by karbon0x**
**Please provide a description of this PR:**
Fixes https://github.com/istio/istio/issues/24169

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
/lgtm

leave this to @howardjohn for approval

File: manifests/charts/istio-control/istio-discovery/files/injection-template.yaml [hunk 1]
```diff
@@ -45,7 +45,11 @@ spec:
     args:
     - istio-iptables
     - "-p"
+    {{- if .MeshConfig.ProxyListenPort }}
+    - "{{ .MeshConfig.ProxyListenPort }}"
+    {{- else }}
     - "15001"
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:51]:
```suggestion
    - {{ .MeshConfig.ProxyListenPort | default "15001" }}
```

may be simpler

**Review by howardjohn — COMMENTED** [review]
lgtm

**Review by howardjohn — APPROVED** [review]

**Pull request merged**